lib/statsd/instrument/backends/udp_backend.rb

```diff
@@ -76,6 +77,11 @@ def generate_packet(metric)
       command
     end
 
+    def packet_tags(tags)
+      tags = tags.select { |tag| self.tags_whitelist.any? { |white| tag.start_with?(white) } } if self.tags_whitelist
```

Comment:
I _think_ this will let through a tag like "hostname" if the whitelist has an entry for "host" because of the `start_with?`.  Probably worth adding a test for this case.

Reply:
Whitelist should have entry for `host:` to prevent this

Reply:
`host` is a special differentiating tag in DD if it is not submitted metrics will still work but if 2 samples from the same metric arrive with the same timestamp without a host tag, last write wins.